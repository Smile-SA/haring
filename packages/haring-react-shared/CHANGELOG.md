# @smile/haring-react-shared

## 0.15.0

### Minor Changes

- [#194](https://github.com/Smile-SA/haring/pull/194) [`97a610a`](https://github.com/Smile-SA/haring/commit/97a610aa02b86423b723ef8ae5d45e8ecf346f92) Thanks [@vapersmile](https://github.com/vapersmile)! - Upgrade mantine version to 7.11.x

## 0.14.0

## 0.13.0

### Minor Changes

- [#178](https://github.com/Smile-SA/haring/pull/178) [`ce8f39a2c52590b2c91950450a84222434913743`](https://github.com/Smile-SA/haring/commit/ce8f39a2c52590b2c91950450a84222434913743) Thanks [@tonai](https://github.com/tonai)! - Rename packages

## 0.12.0

## 0.11.0

## 0.10.0

### Minor Changes

- [#149](https://github.com/Smile-SA/haring/pull/149) [`2706a09`](https://github.com/Smile-SA/haring/commit/2706a097fd86b072cf21d12fe2a97427883671cf) Thanks [@vapersmile](https://github.com/vapersmile)! - Add Calendar style on globlaStyle and add a preview

- [#150](https://github.com/Smile-SA/haring/pull/150) [`34fabfb`](https://github.com/Smile-SA/haring/commit/34fabfbd1a2c048c39adc567548a3ee7e85074ee) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Fixed missing global styles for focus-visible color

## 0.9.0

### Minor Changes

- [`d430e2c`](https://github.com/Smile-SA/haring/commit/d430e2cbcd26871c4db0c53f92a4e4baa6af839c) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Migrated from Mantine V6 to V7

## 0.8.1

## 0.8.0

### Minor Changes

- [#108](https://github.com/Smile-SA/haring/pull/108) [`320634d`](https://github.com/Smile-SA/haring/commit/320634d7ec2048a328cfda1dbcff5d1b70a32563) Thanks [@vapersmile](https://github.com/vapersmile)! - Add custom default style for carousel mantine composent and add 2-Mantine/carousel page

## 0.7.0

### Minor Changes

- [#87](https://github.com/Smile-SA/haring/pull/87) [`dbc06ca`](https://github.com/Smile-SA/haring/commit/dbc06ca55961b69663ab7fdc02609c6525ae361d) Thanks [@tonai](https://github.com/tonai)! - Add theme provider and update barrel file for better next.js compatibility

- [#95](https://github.com/Smile-SA/haring/pull/95) [`5696400`](https://github.com/Smile-SA/haring/commit/5696400e6f703da52db5f7199b50f8251fa76726) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `FilterList` component, renamed `FiltersBar` in `SidebarFilters`
  and `FiltersCheckboxList` in `SearchableCheckboxList`,
  refactored `SearchableCheckboxList`, added common `IFilter` type in shared
  package, added topContent prop to `SwitchableView`, fixed `BrowsingPage` with
  name changes.

## 0.6.0

### Minor Changes

- [#74](https://github.com/Smile-SA/haring/pull/74) [`0ad5377`](https://github.com/Smile-SA/haring/commit/0ad5377535e7c19941da301e8a2ee7298ab70f91) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Refactored actions in `ThumbnailGrid` and `TableGridView` components, moved `typeGuard` helper to shared package, added `typeGuardInterface` function, added tests

## 0.5.0

### Minor Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add dynamic componentProps property in actions

- [#65](https://github.com/Smile-SA/haring/pull/65) [`bd9cfe4`](https://github.com/Smile-SA/haring/commit/bd9cfe42d0b22f5f7f5e7b0de30fdfb22ad3e1c8) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Reworked action/confirm types and moved then in `react-front-kit-shared` then refactored types in `Table`, `Thumbnail` and `ConfirmModal` , Added `ThumbnailGrid` and `TableGridView` components,

- [#68](https://github.com/Smile-SA/haring/pull/68) [`c8b5a39`](https://github.com/Smile-SA/haring/commit/c8b5a3978c8ce7133da0b498d9f0b326f07eb737) Thanks [@QuentinLeCaignec](https://github.com/QuentinLeCaignec)! - Added `storybook-pages` package containing example pages, renamed `TestPage`
  into `BrowsingPage` and completed it with integration of `TableGridView`, fixed
  various component styles and props related to page.

### Patch Changes

- [#72](https://github.com/Smile-SA/haring/pull/72) [`c8d3d94`](https://github.com/Smile-SA/haring/commit/c8d3d946cb0ded4ed26d4b8be249eff45d3d56f6) Thanks [@tonai](https://github.com/tonai)! - Add server barrel files

## 0.4.0

### Minor Changes

- [#53](https://github.com/Smile-SA/haring/pull/53) [`01ee3ea`](https://github.com/Smile-SA/haring/commit/01ee3ea74827cb6e3d4041b50fbab959750135ff) Thanks [@tonai](https://github.com/tonai)! - Use `Paper` for `InfoCard`, update test page and refactor `Pagination` `itemsPerPage` prop

## 0.3.0
