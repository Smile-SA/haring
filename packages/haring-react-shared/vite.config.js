import { resolve } from 'path';

import react from '@vitejs/plugin-react-swc';
import { defineConfig } from 'vite';
import dts from 'vite-plugin-dts';

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.tsx'),
      fileName: '[name]',
      formats: ['cjs', 'es'],
    },
    rollupOptions: {
      external: [
        '@mantine/carousel',
        '@mantine/core',
        '@mantine/dates',
        '@mantine/dropzone',
        '@mantine/hooks',
        '@phosphor-icons/react',
        '@smile/haring-react',
        '@smile/haring-react-dates',
        '@smile/haring-react-dropzone',
        '@smile/haring-react-shared',
        '@smile/haring-react-table',
        '@storybook/addon-actions',
        '@storybook/preview-api',
        '@tabler/icons-react',
        '@testing-library/react',
        'dayjs',
        'embla-carousel-react',
        'mantine-react-table',
        'react',
        'react/jsx-runtime',
      ],
      input: {
        index: resolve(__dirname, 'src/index.tsx'),
        server: resolve(__dirname, 'src/server.tsx'),
        'storybook-utils': resolve(__dirname, 'src/storybook-utils/index.ts'),
        'test-utils': resolve(__dirname, 'src/test-utils/index.ts'),
      },
      output: {
        banner: (chunkInfo) => {
          if (
            ['src/index.tsx'].find((modulePath) =>
              chunkInfo.facadeModuleId?.endsWith(modulePath),
            )
          ) {
            return `"use client"`;
          }
          return '';
        },
        preserveModules: false,
      },
    },
  },
  plugins: [
    react(),
    dts({
      insertTypesEntry: true,
      tsconfigPath: './tsconfig.build.json',
    }),
  ],
});
