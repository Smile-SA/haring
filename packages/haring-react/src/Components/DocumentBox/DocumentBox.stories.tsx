import type { Meta, StoryObj } from '@storybook/react';

import { DocumentBox as Cmp } from './DocumentBox';

const meta = {
  component: Cmp,
  tags: ['autodocs'],
  title: '3-custom/Components/DocumentBox',
} satisfies Meta<typeof Cmp>;

export default meta;
type IStory = StoryObj<typeof meta>;

export const DocumentBox: IStory = {
  args: {
    author: 'Aline',
    children: 'Children',
    date: 'Published on December 24, 2023',
    iconType: 'PDF',
    path: '(Customer > 567890456 > Invoices)',
    title: 'Random_File.PDF',
  },
};
