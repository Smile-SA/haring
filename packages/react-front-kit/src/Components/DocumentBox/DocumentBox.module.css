.author {
  font-weight: 600;
}

.children {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    margin-top: 10px;
  }
  margin-top: 16px;
}

.clickable {
  cursor: pointer;
}

.header {
  display: flex;
}

.headerBottom {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    font-size: 12px;
    margin: 0;
  }
  /*noinspection CssInvalidAtRule*/
  @mixin larger-than $mantine-breakpoint-sm {
    align-items: center;
  }
  color: var(--mantine-color-dark-3);
  display: flex;
  flex-wrap: wrap;
  font-size: 14px;
  font-weight: 400;
  margin-right: 12px;
}

.headerContent {
  display: flex;
  flex-direction: column;
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    gap: 16px;
  }
  /*noinspection CssInvalidAtRule*/
  @mixin larger-than $mantine-breakpoint-sm {
    gap: 4px;
  }
}

.headerTop {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    flex-direction: column;
    margin-left: 40px;
  }
  /*noinspection CssInvalidAtRule*/
  @mixin larger-than $mantine-breakpoint-sm {
    align-items: center;
  }
  display: flex;
  flex-wrap: wrap;
}

.icon {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    height: 24px;
    width: 24px;
  }
  /*noinspection CssInvalidAtRule*/
  @mixin larger-than $mantine-breakpoint-sm {
    height: 38px;
    width: 38px;
  }
}

.iconContainer {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    height: 24px;
    margin: 8.5px 16px 8.5px 0;
    position: absolute;
  }
  margin-right: 16px;
}

.imageRoot {
  border: 1px solid var(--mantine-color-gray-2);
  border-radius: 16px;
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    display: none;
    margin-top: 16px;
    order: 2;
  }
  /*noinspection CssInvalidAtRule*/
  @mixin larger-than $mantine-breakpoint-sm {
    margin-right: 32px;
    max-width: 165px;
    min-height: 164px;
  }
}

.imageRootMobileDisplayed {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    display: initial;
  }
}

.mobileImageButton {
  background: transparent;
  border: 0;
  margin-top: 10px;
  /*noinspection CssInvalidAtRule*/
  @mixin larger-than $mantine-breakpoint-sm {
    display: none;
  }
}

.path {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    font-size: 12px;
  }
  color: var(--mantine-color-dark-6);
  font-weight: 400;
}

.root {
  display: flex;
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    flex-direction: column;
  }
}

.separator {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    margin: 0 7px;
  }
  margin: 0 12px;
}

.title {
  /*noinspection CssInvalidAtRule*/
  @mixin smaller-than $mantine-breakpoint-sm {
    font-size: 14px;
    margin: 0;
  }
  display: flex;
  font-size: 18px;
  font-weight: 700;
  margin: 0 16px 0 0;
}
