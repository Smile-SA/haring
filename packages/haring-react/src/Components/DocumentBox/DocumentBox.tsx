'use client';

import type { ReactNode } from 'react';

import {
  Box,
  Button,
  Image,
  getThemeColor,
  useMantineTheme,
} from '@mantine/core';
import { CaretDown, CaretUp } from '@phosphor-icons/react';
import { FileIcon } from '@smile/haring-react-shared';
import { useState } from 'react';

import defaultImage from '../../../assets/defaultImage.jpg';

import classes from './DocumentBox.module.css';

export interface IDocumentBoxProps {
  altText?: string;
  author?: ReactNode;
  children?: ReactNode;
  date?: ReactNode;
  iconType?: string;
  image?: string;
  mobileImageButtonLabel?: string;
  onCardClick?: () => void;
  path?: ReactNode;
  title?: ReactNode;
}

export function DocumentBox(props: IDocumentBoxProps): ReactNode {
  const {
    altText = 'thumbnail',
    author,
    children,
    date,
    iconType,
    image = defaultImage,
    mobileImageButtonLabel = 'Display preview',
    onCardClick,
    path,
    title,
  } = props;
  const [mobileImageDisplayed, setMobileImageDisplayed] = useState(false);

  const theme = useMantineTheme();

  return (
    <Box
      className={`${classes.root} ${
        onCardClick ? classes.clickable : ''
      } documentBoxRef`}
      onClick={onCardClick}
    >
      <div
        className={`${classes.imageContainer} ${
          mobileImageDisplayed ? classes.imageRootMobileDisplayed : ''
        }`}
      >
        <Image
          alt={altText}
          classNames={{
            root: classes.imageRoot,
          }}
          radius={16}
          src={image}
        />
      </div>
      <div>
        <div>
          <div className={classes.header}>
            <div className={`${classes.iconContainer} documentBoxIconRef`}>
              <FileIcon
                className={classes.icon}
                color={getThemeColor(theme.primaryColor, theme)}
                type={iconType}
                weight="light"
              />
            </div>
            <div className={classes.headerContent}>
              <div className={`${classes.headerTop} documentBoxHeaderTopRef`}>
                {Boolean(title) && (
                  <span className={classes.title}>{title}</span>
                )}
                {Boolean(path) && <span className={classes.path}>{path}</span>}
              </div>
              <div className={classes.headerBottom}>
                {Boolean(date) && (
                  <>
                    <span>{date}</span>
                    {Boolean(author) && (
                      <span className={classes.separator}>-</span>
                    )}
                  </>
                )}
                {Boolean(author) && (
                  <span className={classes.author}>{author}</span>
                )}
              </div>
            </div>
          </div>
        </div>
        <div className={classes.children}>{children}</div>
        <Button
          className={classes.mobileImageButton}
          onClick={() => setMobileImageDisplayed(!mobileImageDisplayed)}
          rightSection={mobileImageDisplayed ? <CaretUp /> : <CaretDown />}
          variant="default"
        >
          {mobileImageButtonLabel}
        </Button>
      </div>
    </Box>
  );
}
