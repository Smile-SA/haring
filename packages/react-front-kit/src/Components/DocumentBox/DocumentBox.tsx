'use client';

import type { ReactNode } from 'react';

import { Box, Button, Image, useMantineTheme } from '@mantine/core';
import { CaretDown, CaretUp } from '@phosphor-icons/react';
import { FileIcon } from '@smile/react-front-kit-shared';
import { useState } from 'react';

import defaultImage from '../../../assets/defaultImage.jpg';

import { useStyles } from './DocumentBox.style';

export interface IDocumentBoxProps {
  author?: ReactNode;
  children?: ReactNode;
  date?: ReactNode;
  iconType?: string;
  image?: string;
  mobileImageButtonLabel?: string;
  onCardClick?: () => void;
  path?: ReactNode;
  title?: ReactNode;
}

export function DocumentBox(props: IDocumentBoxProps): ReactNode {
  const {
    author,
    children,
    date,
    iconType,
    image = defaultImage,
    mobileImageButtonLabel = 'Display preview',
    onCardClick,
    path,
    title,
  } = props;
  const [mobileImageDisplayed, setMobileImageDisplayed] = useState(false);

  const { classes } = useStyles();
  const theme = useMantineTheme();

  return (
    <Box
      className={`${classes.root} ${onCardClick ? classes.clickable : ''}`}
      onClick={onCardClick}
    >
      <Image
        classNames={{
          image: classes.image,
          root: `${classes.imageRoot} ${
            mobileImageDisplayed ? classes.imageRootMobileDisplayed : ''
          }`,
        }}
        radius={16}
        src={image}
      />
      <div>
        <div>
          <div className={classes.header}>
            <div className={classes.iconContainer}>
              <FileIcon
                className={classes.icon}
                color={theme.fn.primaryColor()}
                type={iconType}
                weight="light"
              />
            </div>
            <div className={classes.headerContent}>
              <div className={classes.headerTop}>
                {Boolean(title) && (
                  <span className={classes.title}>{title}</span>
                )}
                {Boolean(path) && <span className={classes.path}>{path}</span>}
              </div>
              <div className={classes.headerBottom}>
                {Boolean(date) && (
                  <>
                    <span>{date}</span>
                    {Boolean(author) && (
                      <span className={classes.separator}>-</span>
                    )}
                  </>
                )}
                {Boolean(author) && (
                  <span className={classes.author}>{author}</span>
                )}
              </div>
            </div>
          </div>
        </div>
        <div className={classes.children}>{children}</div>
        <Button
          className={classes.mobileImageButton}
          onClick={() => setMobileImageDisplayed(!mobileImageDisplayed)}
          rightIcon={mobileImageDisplayed ? <CaretUp /> : <CaretDown />}
          variant="default"
        >
          {mobileImageButtonLabel}
        </Button>
      </div>
    </Box>
  );
}
