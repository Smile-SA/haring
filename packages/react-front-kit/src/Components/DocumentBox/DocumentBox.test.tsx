import { renderWithProviders } from '@smile/haring-react-shared/test-utils';

import { DocumentBox } from './DocumentBox';

describe('DocumentBox', () => {
  it('matches snapshot', () => {
    const { container } = renderWithProviders(
      <DocumentBox
        author="Aline"
        date="Published on December 24, 2023"
        iconType="PDF"
        path="(Customer > 567890456 > Invoices)"
        title="Random_File.PDF"
      >
        children
      </DocumentBox>,
    );
    expect(container).toMatchSnapshot();
  });
});
