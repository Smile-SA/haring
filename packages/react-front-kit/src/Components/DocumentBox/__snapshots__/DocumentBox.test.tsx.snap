// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentBox matches snapshot 1`] = `
<div>
  <div
    class="___ref-documentBox  mantine-1hsxp14"
  >
    <div
      class=" mantine-Image-root mantine-d3it6h"
      style="width: 100%;"
    >
      <figure
        class="mantine-qenwvq mantine-Image-figure"
      >
        <div
          class="mantine-1iugybl mantine-Image-imageWrapper"
        >
          <img
            class="mantine-Image-image mantine-10vaayr"
            src=""
            style="object-fit: cover; width: 100%; height: auto;"
          />
        </div>
      </figure>
    </div>
    <div>
      <div>
        <div
          class="mantine-14n9mxy"
        >
          <div
            class="___ref-documentBoxIcon mantine-cznryd"
          >
            <svg
              class="mantine-wvylvx"
              fill="#0b7285"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M222,152a6,6,0,0,1-6,6H190v20h18a6,6,0,0,1,0,12H190v18a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6h32A6,6,0,0,1,222,152ZM90,172a26,26,0,0,1-26,26H54v10a6,6,0,0,1-12,0V152a6,6,0,0,1,6-6H64A26,26,0,0,1,90,172Zm-12,0a14,14,0,0,0-14-14H54v28H64A14,14,0,0,0,78,172Zm84,8a34,34,0,0,1-34,34H112a6,6,0,0,1-6-6V152a6,6,0,0,1,6-6h16A34,34,0,0,1,162,180Zm-12,0a22,22,0,0,0-22-22H118v44h10A22,22,0,0,0,150,180ZM42,112V40A14,14,0,0,1,56,26h96a6,6,0,0,1,4.25,1.76l56,56A6,6,0,0,1,214,88v24a6,6,0,0,1-12,0V94H152a6,6,0,0,1-6-6V38H56a2,2,0,0,0-2,2v72a6,6,0,0,1-12,0ZM158,82h35.52L158,46.48Z"
              />
            </svg>
          </div>
          <div
            class="mantine-1x591ah"
          >
            <div
              class="___ref-documentBoxHeaderTop mantine-hxvu3z"
            >
              <span
                class="mantine-4yvscz"
              >
                Random_File.PDF
              </span>
              <span
                class="mantine-1g71ejr"
              >
                (Customer &gt; 567890456 &gt; Invoices)
              </span>
            </div>
            <div
              class="mantine-11bkg4o"
            >
              <span>
                Published on December 24, 2023
              </span>
              <span
                class="mantine-1g4onuo"
              >
                -
              </span>
              <span
                class="mantine-l9up18"
              >
                Aline
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mantine-1yhpcgp"
      >
        children
      </div>
      <button
        class="mantine-UnstyledButton-root mantine-Button-root mantine-7bznx3"
        data-button="true"
        type="button"
      >
        <div
          class="mantine-1wpc1xj mantine-Button-inner"
        >
          <span
            class="mantine-1ryt1ht mantine-Button-label"
          >
            Display preview
          </span>
          <span
            class="mantine-Button-icon mantine-Button-rightIcon mantine-467wo8"
          >
            <svg
              fill="currentColor"
              height="1em"
              viewBox="0 0 256 256"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M213.66,101.66l-80,80a8,8,0,0,1-11.32,0l-80-80A8,8,0,0,1,53.66,90.34L128,164.69l74.34-74.35a8,8,0,0,1,11.32,11.32Z"
              />
            </svg>
          </span>
        </div>
      </button>
    </div>
  </div>
</div>
`;
